### PR TITLE
SDI-355 Preview and start migration

### PR DESCRIPTION
**Pull request by andymarke**
Nearly all of incentive migration launch should function fully now, perhaps a small about of work around the migration progress page.

TODO missing a few controller unit tests to complement the cypress test

### CODE REVIEW

File: server/@types/migrationImport/index.d.ts [hunk 1]
```diff
@@ -13,28 +13,48 @@ export interface paths {
   '/queue-admin/purge-queue/{queueName}': {
     put: operations['purgeQueue']
   }
+  '/migrate/visits/{migrationId}/cancel': {
```
Comment (andymarke) [inline server/@types/migrationImport/index.d.ts:16]: All changes auto generated due to swagger changes, no need to review this file

File: server/@types/nomisPrisonerImport/index.d.ts [hunk 2]
```diff
@@ -23,6 +23,14 @@ export interface paths {
     /** Retrieves a paged list of visits by filter */
     get: operations['getVisitsByFilter']
   }
+  '/incentives/ids': {
```
Comment (andymarke) [inline server/@types/nomisPrisonerImport/index.d.ts:26]: All changes auto generated due to swagger changes, no need to review this file

File: server/views/pages/incentives/startIncentivesMigrationPreview.njk [hunk 3]
```diff
@@ -0,0 +1,128 @@
+{% extends "../../partials/layout.njk" %}
```
Comment (andymarke) [inline server/views/pages/incentives/startIncentivesMigrationPreview.njk:1]: might look at extracting nunchuck templates - all these pages are very similar. It, however given they are also very transient (e.g not need after final migration) it might just not be worth it

File: package-lock.json [hunk 4]
```diff
@@ -12151,9 +12151,9 @@
       }
     },
     "node_modules/typescript": {
-      "version": "4.7.4",
-      "resolved": "https://registry.npmjs.org/typescript/-/typescript-4.7.4.tgz",
-      "integrity": "sha512-C0WQT0gezHuw6AdY1M2jxUO83Rjf0HP7Sk1DtXj6j1EwkQNZrHAg2XPWlq62oqEhYvONq5pkC2Y9oPljWToLmQ==",
+      "version": "4.8.2",
```
Comment (andymarke) [inline package-lock.json:12154]: unrelated fix to update warning

File: server/@types/migrationImport/index.d.ts [hunk 5]
```diff
@@ -226,6 +280,37 @@ export interface operations {
       }
     }
   }
+  /** Requires role <b>MIGRATE_VISITS</b> */
+  cancel: {
```
Comment (lbennett-moj) [inline server/@types/migrationImport/index.d.ts:284]: incentives role?

File: server/routes/incentiveMigration/incentivesMigrationController.ts [hunk 6]
```diff
@@ -90,10 +96,43 @@ export default class IncentivesMigrationController {
...
   }
 
+  async startIncentiveMigrationPreview(req: Request, res: Response): Promise<void> {
+    res.render('pages/incentives/startIncentivesMigrationPreview', { form: req.session.startIncentivesMigrationForm })
+  }
+
+  async postClearDLQIncentiveMigrationPreview(req: Request, res: Response): Promise<void> {
+    const result = await this.nomisMigrationService.deleteIncentivesFailures(context(res))
+    logger.info(`${result.messagesFoundCount} failures deleted`)
+    req.body = { ...req.session.startVisitsMigrationForm }
```
Comment (lbennett-moj) [inline server/routes/incentiveMigration/incentivesMigrationController.ts:117]: wrong form?

File: server/@types/migrationImport/index.d.ts (hunk 5 above)
Comment (lbennett-moj) [reply server/@types/migrationImport/index.d.ts:284]: sorry ignore - needs correcting in the nomis api

File: server/routes/incentiveMigration/incentivesMigrationController.ts (hunk 6 above)
Comment (andymarke) [reply server/routes/incentiveMigration/incentivesMigrationController.ts]: Yes, corrected. Thanks

**Review by lbennett-moj — APPROVED** [review]
